Tools/cases_generator/interpreter_definition.md

```diff
@@ -142,9 +142,9 @@ The following definitions may occur:
 The optional `type` in an `object` is the C type. It defaults to `PyObject *`.
 The objects before the "--" are the objects on top of the stack at the start of
 the instruction. Those after the "--" are the objects on top of the stack at the
-end of the instruction. When prefixed by a `&`, the type rule follows the
-`type_prop` rule. This indicates the stack effect is of that specific type
-after the operation. In this case, the type may also contain auxillary information
+end of the instruction. When prefixed by a `&`, the `type` production rule follows the
+`type_prop` production rule. This indicates the stack effect is of that specific type
+after the operation. In this case, the type may also consist of auxillary information
```

Comment:
"contain" was probably more correct. The problem is "auxillary information".
How about "in addition to the type, a refinement may be added (`type + refinement`), which must be one of the following:"
Then list the types here, rather than having them in an appendix.